### PR TITLE
Seller Experience: Use Headstart to set up the payments pattern

### PR DESCRIPTION
**Pull request by sixhours**
#### Changes proposed in this Pull Request

* During the API callback for Store Features, add a call to change the theme to Zoologist, then hit `theme-setup` to run Headstart on the theme.
* Move design selection step earlier in the flow so it's not the next step after Store Features. This way the other flows still work, but Store Features has no "next step" according to the flow.
* @TODO: Figure out how to use a different annotation from the theme default -- possible API work needed

#### Testing instructions

* TBD

Related to #60731

### CODE REVIEW

File: client/lib/signup/step-actions/index.js [hunk 1]
```diff
@@ -437,6 +437,35 @@ export function setOptionsOnSite( callback, { siteSlug, siteTitle, tagline } ) {
...
 
+export function setStoreFeatures( callback, { siteSlug } ) {
+	if ( ! siteSlug ) {
+		defer( callback );
+		return;
+	}
+
+	wpcom.req
+		.post( `/sites/${ siteSlug }/themes/mine`, { theme: 'zoologist', dont_change_homepage: true } )
+		.then( () =>
```
Comment (sixhours) [inline client/lib/signup/step-actions/index.js:449]: I don't think the theme setup is working here. If I run Headstart on a new site using Zoologist, I get different content than what I get when you sign up with this patch.

Comment (sixhours) [reply client/lib/signup/step-actions/index.js:449]: Aha, my last merge messed this up. It appears to be working now

**Comment by sixhours** [comment]
Opened a diff to update the `theme-setup` endpoint with an arg specific to store requests: D74664-code

**Review by blackjackkent — APPROVED** [review]
Ran through all the intent flows and everything seems to be clean. Merging this so Nelson can build on it.

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/7143573

Hi @sixhours, could you please edit the description of this PR and add a screenshot for our translators? Ideally it'd include this string: `Describe your item. Use this section to add a full description and details of your product, along with its many selling points.`

Thank you in advance!

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/7143573).

**Pull request merged**